chia/data_layer/data_store.py

```diff
@@ -1254,13 +1268,27 @@ async def insert_batch(
     ) -> Optional[bytes32]:
         async with self.db_wrapper.writer() as writer:
             old_root = await self.get_tree_root(tree_id)
-            root_hash = old_root.node_hash
-            if old_root.node_hash is None:
+            pending_root = await self.get_pending_root(tree_id=tree_id)
+            if pending_root is None:
+                intermediate_root: Optional[Root] = old_root
+            else:
+                if pending_root.status == Status.PENDING_BATCH:
+                    # We have an unfinished batch, continue the current batch on top of it.
+                    if pending_root.generation != old_root.generation + 1:
+                        raise Exception("Internal error")
+                    await self.change_root_status(pending_root, Status.COMMITTED)
```

Comment:
This feels misleading as it is not committed on chain, it is merely not the latest pending (uncommited local, whatever term) root.  I don't know what all functional ramifications there are around this presently, but it seems like clearling 'pending' roots wouldn't work correctly with this.  Well, you wouldn't clear back to the latest on-chain root.